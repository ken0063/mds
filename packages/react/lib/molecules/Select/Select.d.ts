import { FC, ReactNode } from 'react';
export declare const KEY_CODES: {
    ENTER: number;
    SPACE: number;
    DOWN_ARROW: number;
    ESC: number;
    UP_ARROW: number;
};
interface SelectOptionProps {
    value: string;
    label: string;
}
interface RenderOptionProps {
    isSelected: boolean;
    options: SelectOptionProps;
    getOptionRecommendedProps: (overrideProps?: Object) => Object;
}
interface SelectProps {
    optionHandler?: (option: SelectOptionProps, optionIndex: number) => void;
    options?: SelectOptionProps[];
    label?: string;
    renderOption?: (props: RenderOptionProps) => ReactNode;
}
declare const Select: FC<SelectProps>;
export default Select;
