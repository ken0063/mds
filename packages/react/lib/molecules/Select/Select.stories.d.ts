import "@mds/scss/lib/Select.css";
declare const _default: {
    title: string;
    decorators: (() => core_dist_types.ProjectAnnotations<core_dist_types.Renderer>)[];
};
export default _default;
export declare const Common: () => import("react/jsx-runtime").JSX.Element;
export declare const RenderOption: () => import("react/jsx-runtime").JSX.Element;
export declare const CustomLabel: () => import("react/jsx-runtime").JSX.Element;
