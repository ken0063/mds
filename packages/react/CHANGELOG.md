# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.1.0 (2025-07-01)


### Bug Fixes

* **ci:** fix failing pipeline ([a6fd456](https://github.com/ken0063/mds/commit/a6fd4564230514fa44264d272334e7d08bb8b5f2))
* **docs:** updated comments in packages ([0c08414](https://github.com/ken0063/mds/commit/0c084144364466831c2ec6f562fb4eeb61cd8d37))


### Features

* **margin componenet:** added magin stories ([e31f3bc](https://github.com/ken0063/mds/commit/e31f3bc2b99ee3b636d37cb014431809615f7145))
