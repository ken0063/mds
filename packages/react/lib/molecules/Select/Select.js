import React, { useState, useRef, useEffect, createRef } from 'react';
import Text from '../../atoms/Text/Text.js';

const KEY_CODES = {
    ENTER: 13,
    SPACE: 32,
    DOWN_ARROW: 40,
    ESC: 27,
    UP_ARROW: 38,
};
const getPreviousOptionIndex = (currentIndex, options) => {
    if (currentIndex === null) {
        return 0;
    }
    if (currentIndex === 0) {
        return options.length - 1;
    }
    return currentIndex - 1;
};
const getNextOptionIndex = (currentIndex, options) => {
    if (currentIndex === null) {
        return 0;
    }
    if (currentIndex === options.length - 1) {
        return 0;
    }
    return currentIndex + 1;
};
const Select = ({ options = [], label = 'Please select an option', optionHandler, renderOption, }) => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedIndex, setSelectedIndex] = useState(null);
    const [highlightedIndex, setHighlightedIndex] = useState(null);
    const labelRef = useRef(null);
    const [optionRefs, setOptionRefs] = useState([]);
    const [overlayTop, setOverlayTop] = useState(0);
    const handleSelectToggle = () => {
        setIsOpen(!isOpen);
    };
    const onOptionSelected = (option, optionIndex) => {
        if (optionHandler) {
            optionHandler(option, optionIndex);
        }
        setSelectedIndex(optionIndex);
        setIsOpen(false);
    };
    useEffect(() => {
        setOverlayTop(labelRef.current?.offsetHeight || 0);
    }, []);
    let selectedOption = null;
    if (selectedIndex !== null) {
        selectedOption = options[selectedIndex];
    }
    const highlightOption = (optionIndex) => {
        setHighlightedIndex(optionIndex);
    };
    const onButtonKeyDown = (event) => {
        event.preventDefault();
        if ([KEY_CODES.ENTER, KEY_CODES.SPACE, KEY_CODES.DOWN_ARROW].includes(event.keyCode)) {
            setIsOpen(true);
            // set focus on the list item
            highlightOption(0);
        }
    };
    useEffect(() => {
        setOptionRefs(options.map((_) => createRef()));
    }, [options.length]);
    useEffect(() => {
        if (highlightedIndex !== null && isOpen) {
            const ref = optionRefs[highlightedIndex];
            if (ref && ref.current) {
                ref.current.focus();
            }
        }
    }, [isOpen, highlightedIndex]);
    const onOptionKeyDown = (e) => {
        if (e.keyCode === KEY_CODES.ESC) {
            setIsOpen(false);
            return;
        }
        if (e.keyCode === KEY_CODES.DOWN_ARROW) {
            highlightOption(getNextOptionIndex(highlightedIndex, options));
        }
        if (e.keyCode === KEY_CODES.UP_ARROW) {
            highlightOption(getPreviousOptionIndex(highlightedIndex, options));
        }
        if (e.keyCode === KEY_CODES.ENTER) {
            onOptionSelected(options[highlightedIndex], highlightedIndex);
        }
    };
    return (React.createElement("div", { className: "mds-select" },
        React.createElement("button", { ref: labelRef, "data-testid": "mds-select-button", className: "mds-select__label", onClick: handleSelectToggle, "aria-controls": "mds-select-list", "aria-haspopup": true, "aria-expanded": isOpen ? true : undefined, onKeyDown: onButtonKeyDown },
            React.createElement(Text, null, selectedOption === null ? label : selectedOption?.label),
            React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 24 24", strokeWidth: "1.5", stroke: "currentColor", width: "1rem", height: "1rem", className: `mds-select__caret ${isOpen ? 'mds-select__caret--open' : 'mds-select__caret--close'}` },
                React.createElement("path", { strokeLinecap: "round", strokeLinejoin: "round", d: "M19.5 8.25l-7.5 7.5-7.5-7.5" }))),
        React.createElement("ul", { role: "menu", "aria-hidden": isOpen ? undefined : false, id: "mds-select-list", className: `mds-select__overlay ${isOpen ? 'mds-select__overlay--open' : ''}`, style: { top: overlayTop } }, options?.map((item, idx) => {
            const isSelected = selectedIndex === idx;
            const isHighlighted = highlightedIndex === idx;
            const ref = optionRefs[idx];
            const renderOptionProps = {
                ref,
                options: item,
                isSelected,
                getOptionRecommendedProps: (overrideProps = {}) => {
                    return {
                        ref,
                        role: 'menuitemradio',
                        'aria-label': item?.label,
                        'aria-checked': isSelected ? true : undefined,
                        onKeyDown: onOptionKeyDown,
                        tabIndex: isHighlighted ? -1 : 0,
                        onMouseEnter: () => highlightOption(idx),
                        onMouseLeave: () => highlightOption(null),
                        className: `mds-select__option
                        ${isSelected ? 'mds-select__option--selected' : ''}
                        ${isHighlighted ? 'mds-select__option--highlighted' : ''}
                    `,
                        key: item?.value,
                        onClick: () => onOptionSelected(item, idx),
                        ...overrideProps,
                    };
                },
            };
            if (renderOption) {
                return renderOption(renderOptionProps);
            }
            return (React.createElement("li", { ...renderOptionProps.getOptionRecommendedProps() },
                React.createElement(Text, null, item?.label),
                isSelected && (React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", fill: "none", viewBox: "0 0 24 24", strokeWidth: "1.5", stroke: "currentColor", width: "1rem", height: "1rem" },
                    React.createElement("path", { strokeLinecap: "round", strokeLinejoin: "round", d: "M4.5 12.75l6 6 9-13.5" })))));
        }))));
};

export { KEY_CODES, Select as default };
//# sourceMappingURL=Select.js.map
