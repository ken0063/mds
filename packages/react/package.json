{
  "name": "@mds/react",
  "version": "1.1.0",
  "main": "index.js",
  "license": "MIT",
  "type": "module",
  "files": [
    "lib"
  ],
  "devDependencies": {
    "@babel/core": "^7.27.7",
    "@babel/preset-env": "^7.27.2",
    "@babel/preset-react": "^7.27.1",
    "@babel/preset-typescript": "^7.27.1",
    "@rollup/plugin-commonjs": "^28.0.6",
    "@rollup/plugin-node-resolve": "^16.0.1",
    "@rollup/plugin-typescript": "^12.1.4",
    "@storybook/addon-a11y": "^9.0.15",
    "@storybook/addon-actions": "^9.0.8",
    "@storybook/addon-essentials": "^8.6.14",
    "@storybook/addon-interactions": "^8.6.14",
    "@storybook/addon-knobs": "^8.0.1",
    "@storybook/addon-links": "^9.0.15",
    "@storybook/builder-webpack4": "^6.5.16",
    "@storybook/manager-webpack4": "^6.5.16",
    "@storybook/react": "^9.0.15",
    "@storybook/testing-library": "^0.2.2",
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.3.0",
    "@types/jest": "^30.0.0",
    "@types/react": "^19.1.8",
    "acorn-jsx": "^5.3.2",
    "babel-loader": "^10.0.0",
    "chromatic": "^13.0.1",
    "jest": "^30.0.3",
    "jest-environment-jsdom": "^30.0.2",
    "react": "^19.1.0",
    "rollup": "^4.44.1",
    "rollup-plugin-typescript2": "^0.36.0",
    "typescript": "^5.8.3"
  },
  "scripts": {
    "build": "rollup -c ",
    "dev": "yarn build --watch",
    "test": "jest --verbose --env=jsdom",
    "test:watch": "yarn test --watch",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook",
    "test:chromatic": "npx chromatic --project-token=\"$CHROMATIC_PROJECT_TOKEN\""
  },
  "publishConfig": {
    "access": "public"
  },
  "readme": "ERROR: No README data found!",
  "_id": "@mds/react@1.0.0",
  "dependencies": {
    "@mds/foundation": "workspace:^",
    "tslib": "^2.8.1"
  }
}
