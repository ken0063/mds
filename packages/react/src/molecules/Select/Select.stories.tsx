import React from 'react';

import Select from './Select';
import { withA11y } from '@storybook/addon-a11y';

import '@mds/scss/lib/Select.css';

export default {
  title: 'Molecules/Select',
  decorators: [withA11y],
};

const options = [
  {
    label: 'bda hs',
    value: 'shbavjs',
  },
  {
    label: 'bsahc',
    value: 'shbsmhavjs',
  },
  {
    label: 'jkhasda',
    value: 'sjksbcavjs',
  },
];

export const Common = () => <Select options={options} />;

export const RenderOption = () => (
  <Select
    options={options}
    renderOption={({ options, getOptionRecommendedProps, isSelected }) => {
      return (
        <span {...getOptionRecommendedProps()}>
          {options?.label} {isSelected ? 'SELECTED !!!' : ''}
        </span>
      );
    }}
  />
);

export const CustomLabel = () => (
  <Select options={options} label="Select a color" />
);
