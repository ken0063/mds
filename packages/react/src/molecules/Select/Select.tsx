import React, {
  createRef,
  FC,
  KeyboardEventHandler,
  ReactNode,
  useEffect,
  useRef,
  useState,
} from 'react';
import Text from '../../atoms/Text';

export const KEY_CODES = {
  ENTER: 13,
  SPACE: 32,
  DOWN_ARROW: 40,
  ESC: 27,
  UP_ARROW: 38,
};

interface SelectOptionProps {
  value: string;
  label: string;
}
interface RenderOptionProps {
  isSelected: boolean;
  options: SelectOptionProps;
  getOptionRecommendedProps: (overrideProps?: Object) => Object;
}

interface SelectProps {
  optionHandler?: (option: SelectOptionProps, optionIndex: number) => void;
  options?: SelectOptionProps[];
  label?: string;
  renderOption?: (props: RenderOptionProps) => ReactNode;
}

const getPreviousOptionIndex = (
  currentIndex: number | null,
  options: Array<SelectOptionProps>
) => {
  if (currentIndex === null) {
    return 0;
  }

  if (currentIndex === 0) {
    return options.length - 1;
  }

  return currentIndex - 1;
};

const getNextOptionIndex = (
  currentIndex: number | null,
  options: Array<SelectOptionProps>
) => {
  if (currentIndex === null) {
    return 0;
  }

  if (currentIndex === options.length - 1) {
    return 0;
  }

  return currentIndex + 1;
};

const Select: FC<SelectProps> = ({
  options = [],
  label = 'Please select an option',
  optionHandler,
  renderOption,
}) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const [selectedIndex, setSelectedIndex] = useState<null | number>(null);
  const [highlightedIndex, setHighlightedIndex] = useState<null | number>(null);
  const labelRef = useRef<HTMLButtonElement>(null);
  const [optionRefs, setOptionRefs] = useState<
    React.RefObject<HTMLLIElement>[]
  >([]);
  const [overlayTop, setOverlayTop] = useState<number>(0);

  const handleSelectToggle = () => {
    setIsOpen(!isOpen);
  };

  const onOptionSelected = (option: SelectOptionProps, optionIndex: number) => {
    if (optionHandler) {
      optionHandler(option, optionIndex);
    }
    setSelectedIndex(optionIndex);
    setIsOpen(false);
  };

  useEffect(() => {
    setOverlayTop(labelRef.current?.offsetHeight || 0);
  }, []);

  let selectedOption = null;
  if (selectedIndex !== null) {
    selectedOption = options[selectedIndex];
  }

  const highlightOption = (optionIndex: number | null) => {
    setHighlightedIndex(optionIndex);
  };

  const onButtonKeyDown: KeyboardEventHandler = (event) => {
    event.preventDefault();

    if (
      [KEY_CODES.ENTER, KEY_CODES.SPACE, KEY_CODES.DOWN_ARROW].includes(
        event.keyCode
      )
    ) {
      setIsOpen(true);

      // set focus on the list item
      highlightOption(0);
    }
  };

  useEffect(() => {
    setOptionRefs(options.map((_) => createRef<HTMLLIElement>()));
  }, [options.length]);

  useEffect(() => {
    if (highlightedIndex !== null && isOpen) {
      const ref = optionRefs[highlightedIndex];

      if (ref && ref.current) {
        ref.current.focus();
      }
    }
  }, [isOpen, highlightedIndex]);

  const onOptionKeyDown: KeyboardEventHandler = (e) => {
    if (e.keyCode === KEY_CODES.ESC) {
      setIsOpen(false);

      return;
    }

    if (e.keyCode === KEY_CODES.DOWN_ARROW) {
      highlightOption(getNextOptionIndex(highlightedIndex, options));
    }

    if (e.keyCode === KEY_CODES.UP_ARROW) {
      highlightOption(getPreviousOptionIndex(highlightedIndex, options));
    }

    if (e.keyCode === KEY_CODES.ENTER) {
      onOptionSelected(options[highlightedIndex!], highlightedIndex!);
    }
  };

  return (
    <div className="mds-select">
      <button
        ref={labelRef}
        data-testid="mds-select-button"
        className="mds-select__label"
        onClick={handleSelectToggle}
        aria-controls="mds-select-list"
        aria-haspopup={true}
        aria-expanded={isOpen ? true : undefined}
        onKeyDown={onButtonKeyDown}
      >
        <Text>{selectedOption === null ? label : selectedOption?.label}</Text>

        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          strokeWidth="1.5"
          stroke="currentColor"
          width="1rem"
          height="1rem"
          className={`mds-select__caret ${
            isOpen ? 'mds-select__caret--open' : 'mds-select__caret--close'
          }`}
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          />
        </svg>
      </button>

      <ul
        role="menu"
        aria-hidden={isOpen ? undefined : false}
        id="mds-select-list"
        className={`mds-select__overlay ${
          isOpen ? 'mds-select__overlay--open' : ''
        }`}
        style={{ top: overlayTop }}
      >
        {options?.map((item, idx) => {
          const isSelected = selectedIndex === idx;
          const isHighlighted = highlightedIndex === idx;
          const ref = optionRefs[idx];
          const renderOptionProps = {
            ref,
            options: item,
            isSelected,
            getOptionRecommendedProps: (overrideProps = {}) => {
              return {
                ref,
                role: 'menuitemradio',
                'aria-label': item?.label,
                'aria-checked': isSelected ? true : undefined,
                onKeyDown: onOptionKeyDown,
                tabIndex: isHighlighted ? -1 : 0,
                onMouseEnter: () => highlightOption(idx),
                onMouseLeave: () => highlightOption(null),
                className: `mds-select__option
                        ${isSelected ? 'mds-select__option--selected' : ''}
                        ${
                          isHighlighted ? 'mds-select__option--highlighted' : ''
                        }
                    `,
                key: item?.value,
                onClick: () => onOptionSelected(item, idx),
                ...overrideProps,
              };
            },
          };
          if (renderOption) {
            return renderOption(renderOptionProps);
          }
          return (
            <li {...renderOptionProps.getOptionRecommendedProps()}>
              <Text>{item?.label}</Text>
              {isSelected && (
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  strokeWidth="1.5"
                  stroke="currentColor"
                  width="1rem"
                  height="1rem"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    d="M4.5 12.75l6 6 9-13.5"
                  />
                </svg>
              )}
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export default Select;
