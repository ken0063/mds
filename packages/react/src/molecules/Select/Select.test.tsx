import React from 'react';
import Select from './Select';

import { render, fireEvent } from '@testing-library/react';

/**
 * @jest-environment jsdom
 */
const options = [
  {
    label: 'bda hs',
    value: 'shbavjs',
  },
  {
    label: 'bsahc',
    value: 'shbsmhavjs',
  },
  {
    label: 'jkhasda',
    value: 'sjksbcavjs',
  },
];

test('render all options passed to it', () => {
  const { getAllByRole, getByTestId } = render(<Select options={options} />);
  fireEvent.click(getByTestId('mds-select-button'));
  expect(getAllByRole('menuitemradio')).toHaveLength(options.length);
});

test('render options using custom renderOption method if passed as prop', () => {
  const { getAllByTestId, getByTestId } = render(
    <Select
      options={options}
      renderOption={({ options, getOptionRecommendedProps }) => {
        return (
          <li
            data-testid="custom-render-option"
            {...getOptionRecommendedProps()}
          >
            {options?.label}
          </li>
        );
      }}
    />
  );

  fireEvent.click(getByTestId('mds-select-button'));
  expect(getAllByTestId('custom-render-option')).toHaveLength(options.length);
});

test('calls the onOptionSelected prop with the selected option and its index if passed', () => {
  const optionHandler = jest.fn();
  const { getAllByRole, getByTestId } = render(
    <Select options={options} optionHandler={optionHandler} />
  );

  fireEvent.click(getByTestId('mds-select-button'));
  fireEvent.click(getAllByRole('menuitemradio')[0]);
  expect(optionHandler).toHaveBeenCalledWith(options[0], 0);
});

test('the button label changes to the selected option label', () => {
  const { getAllByRole, getByTestId } = render(<Select options={options} />);

  fireEvent.click(getByTestId('mds-select-button'));
  fireEvent.click(getAllByRole('menuitemradio')[0]);

  expect(getByTestId('mds-select-button')).toHaveTextContent(options[0].label);
});

test('snapshot of the selected option state', () => {
  const { getAllByRole, getByTestId, asFragment } = render(
    <Select options={options} />
  );

  fireEvent.click(getByTestId('mds-select-button'));
  fireEvent.click(getAllByRole('menuitemradio')[0]);

  expect(asFragment()).toMatchSnapshot();
});

test('snapshot of the base state', () => {
  const { asFragment } = render(<Select options={options} />);

  expect(asFragment()).toMatchSnapshot();
});

test('snapshot of the selected option state', () => {
  const { getByTestId, asFragment } = render(<Select options={options} />);

  fireEvent.click(getByTestId('mds-select-button'));

  expect(asFragment()).toMatchSnapshot();
});

test('can customize select label', () => {
  const { getByText } = render(
    <Select options={options} label="Hello World!!!" />
  );

  expect(getByText(/Hello World!!!/)).toBeInTheDocument();
});
