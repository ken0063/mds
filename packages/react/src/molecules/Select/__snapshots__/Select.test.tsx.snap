// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <div
    class="mds-select"
  >
    <button
      aria-controls="mds-select-list"
      aria-haspopup="true"
      class="mds-select__label"
      data-testid="mds-select-button"
    >
      <p
        class="mds-text mds-text-base"
      >
        Please select an option
      </p>
      <svg
        class="mds-select__caret mds-select__caret--close"
        fill="none"
        height="1rem"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot of the selected option state 1`] = `
<DocumentFragment>
  <div
    class="mds-select"
  >
    <button
      aria-controls="mds-select-list"
      aria-haspopup="true"
      class="mds-select__label"
      data-testid="mds-select-button"
    >
      <p
        class="mds-text mds-text-base"
      >
        bda hs
      </p>
      <svg
        class="mds-select__caret mds-select__caret--close"
        fill="none"
        height="1rem"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot of the selected option state 2`] = `
<DocumentFragment>
  <div
    class="mds-select"
  >
    <button
      aria-controls="mds-select-list"
      aria-expanded="true"
      aria-haspopup="true"
      class="mds-select__label"
      data-testid="mds-select-button"
    >
      <p
        class="mds-text mds-text-base"
      >
        Please select an option
      </p>
      <svg
        class="mds-select__caret mds-select__caret--open"
        fill="none"
        height="1rem"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        width="1rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
    <ul
      class="mds-select__overlay"
      id="mds-select-list"
      role="menu"
      style="top: 0px;"
    >
      <li
        aria-label="bda hs"
        class="mds-select__option
                        
                        
                    "
        role="menuitemradio"
        tabindex="0"
      >
        <p
          class="mds-text mds-text-base"
        >
          bda hs
        </p>
      </li>
      <li
        aria-label="bsahc"
        class="mds-select__option
                        
                        
                    "
        role="menuitemradio"
        tabindex="0"
      >
        <p
          class="mds-text mds-text-base"
        >
          bsahc
        </p>
      </li>
      <li
        aria-label="jkhasda"
        class="mds-select__option
                        
                        
                    "
        role="menuitemradio"
        tabindex="0"
      >
        <p
          class="mds-text mds-text-base"
        >
          jkhasda
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
